cmd/utils.go

```diff
@@ -229,3 +230,25 @@ func toS3ETag(etag string) string {
 
 	return etag
 }
+
+// Returns "/bucketName/objectName" for path-style or virtual-host-style requests.
+func getResource(path string, host string, domain string) (string, error) {
+	if domain == "" {
+		return path, nil
+	}
+
+	// If virtual-host-style is enabled construct the "resource" properly.
+	if strings.Contains(host, ":") {
+		// In bucket.mydomain.com:9000, strip out :9000
+		var err error
+		if host, _, err = net.SplitHostPort(host); err != nil {
+			errorIf(err, "Unable to split %s", host)
+			return "", err
+		}
+	}
+	if !strings.HasSuffix(host, "."+domain) {
+		return path, nil
+	}
+	bucket := strings.TrimSuffix(host, "."+domain)
```

Comment:
I think we can validate the bucket name here as soon as we extract it using `isValidBucketName()`

Reply:
Better leave this validation to object layer. Because object layer can better control what kind of bucket names are allowed. For example Azure does not allow bucket names with `.` but aws-s3 allows